src/ParallelAlgorithms/Events/ObserveTimeStep.hpp

```diff
@@ -0,0 +1,182 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <pup.h>
+#include <pup_stl.h>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "IO/Observer/ArrayComponentId.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"  // IWYU pragma: keep
+#include "IO/Observer/ReductionActions.hpp"   // IWYU pragma: keep
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ArrayIndex.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Time/Slab.hpp"
+#include "Time/Time.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Registration.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Tags {
+struct Time;
+struct TimeStep;
+}  // namespace Tags
+/// \endcond
+
+namespace Events {
+template <typename Metavariables, typename EventRegistrars>
+class ObserveTimeStep;
+
+namespace Registrars {
+template <typename Metavariables>
+using ObserveTimeStep =
+    ::Registration::Registrar<Events::ObserveTimeStep, Metavariables>;
+}  // namespace Registrars
+
+/*!
+ * \brief %Observe the size of the time steps.
+ *
+ * Writes reduction quantities:
+ * - `%Time`
+ * - `NumberOfPoints`
+ * - `%Slab size`
+ * - `Minimum time step`
+ * - `Maximum time step`
+ * - `Effective time step`
+ *
+ * The effective time step is the step size of a global-time-stepping
+ * method that would perform a similar amount of work.
```

Comment:
would you mind noting what this measure is in terms of an equation? It looks like its 
```
1 / ( (sum_{elements} pts_per_element / element_step_size) / total_points)
```
if I've followed the code correctly